src/membership/members.rs

```diff
@@ -70,6 +98,8 @@ pub struct Profile<T: Trait> {
     pub entry: EntryMethod<T>,
     pub suspended: bool,
     pub subscription: Option<T::SubscriptionId>,
+    pub controller_account: T::AccountId,
+    pub roles: BTreeMap<RoleId, Vec<ActorId>>,
```

Comment:
Would it be more accurate to have `BTreeMap<RoleId, BTreeMap<ActordId,()> >` as the type here, to enforce that the same actor id cannot exist multiple times under the same role id? This presumes `BTreeMap` is iterable, which I suspect it is.

Reply:
Yes BTreeMap is Iterable.
I agree, effectively using it as a Set, because that's what we really want to represent and it gives us the type protection to enforce it. I'm just concerned with the complexity of this type and weather we can dervie Encode/Decode traits for it. Will have to check.

If its not possible we can always implement a `Set` backed by a `Vec`.

Reply:
Since @siman pointed out that there is a `[BTreeSet](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html)` we can have:

`pub roles: BTreeMap<RoleId, BTreeSet<ActorId>>` 

Not sure about the `Codec` support in substrate and polkadot-js for such a type.

cc: @yourheropaul

Reply:
Agreed, the current solution `ActorInRoleSet<T::ActorId>` looks fine.